src/reporters/udp_sender.js

```diff
@@ -126,8 +126,8 @@ export default class UDPSender {
         );
 
         if (writeResult.err) {
-            this._logger.error(`error writing Thrift object: ${writeResult.err}`);
-            return {err: true, numSpans: numSpans};
+            let errMessage = this._logger.error(`error writing Thrift object: ${writeResult.err}`);
```

Comment:
logger doesn't return a string afaik. Can you add a unit test that verifies that the error is returned as a string?

Reply:
At the very least, the `NullLogger` doesn't:

https://github.com/jaegertracing/jaeger-client-node/blob/8b783aaf26ac007d2f064ad72904d91fddcf020a/src/logger.js#L14-L16